sdks/java/core/src/main/java/org/apache/beam/sdk/io/Sink.java

```diff
@@ -259,6 +259,15 @@ public abstract void finalize(Iterable<WriteT> writerResults, PipelineOptions op
     public abstract WriteT close() throws Exception;
 
     /**
+     * Aborts writing the bundle. Removes resources used for writing the bundle,
+     * and closes this {@link Writer}.
+     *
+     * <p>Called for each failed bundle with best effort. However, there is no strong guarantee in
```

Comment:
I would reword this to something more like

"When a bundle fails, the runner will `abort` the Writer if possible."
and link to `@Teardown` as a reference.

I don't think we need to give examples of when that will occur - just that we'll try but can't always succeed. It may be worth also noting that additional cleanup can be done as part of the `finalize` step in case `abort` could not be called on all failed bundles.